### PR TITLE
[Admin] Locale feature

### PR DESCRIPTION
**Pull request by lchrusciel**
| Q               | A
| --------------- | ---
| Bug fix?        | no
| New feature?    | no
| BC breaks?      | no
| Deprecations?   | no
| Related tickets | 
| License         | MIT
| Doc PR          | 
Based on #4444,  #4395 and #4467

### CODE REVIEW

File: features/localization/managing_locales/adding_locale.feature [hunk 1]
```diff
@@ -0,0 +1,16 @@
+@addressing
+Feature: Adding locale
+    In order to present my goods in different languages
```
Comment (pjedrzejewski) [inline features/localization/managing_locales/adding_locale.feature]: This is not only about products, so I'd say something like:

```
In order to provide service to my customers in their preferred language
```

Opinions?

**Comment by pjedrzejewski** [comment]
@lchrusciel To keep things simple, I think you can add all scenarios in a single PR.

**Comment by lchrusciel** [comment]
:+1:

**Comment by lchrusciel** [comment]
Based on #4444

**Comment by lchrusciel** [comment]
And based on #4395 also

**Comment by pamil** [comment]
@lchrusciel it would be more readable if all "based on..." were in the opening post.

**Comment by lchrusciel** [comment]
@pamil :+1:

File: src/Sylius/Behat/spec/Context/Ui/Admin/ManagingLocaleContextSpec.php [hunk 2]
```diff
@@ -0,0 +1,101 @@
...
+    {
+        $this->beConstructedWith($indexPage, $createPage);
+    }
+
+    function it_is_initializable()
+    {
+        $this->shouldHaveType('Sylius\Behat\Context\Ui\Admin\ManagingLocaleContext');
+    }
+
+    function it_implements_context()
```
Comment (Zales0123) [inline src/Sylius/Behat/spec/Context/Ui/Admin/ManagingLocaleContextSpec.php]: ``it_implements_context_interface`` or ``it_is_context``?

File: features/localization/managing_locales/adding_locale.feature [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+    In order to provide service to my customers in their preferred language
+    As an Administrator
+    I want to add locale to the registry
+
+    Background:
+        Given I am logged in as administrator
+
+    @ui
+    Scenario: Adding new locale
+        Given I want to create new locale
```
Comment (TheMadeleine) [inline features/localization/managing_locales/adding_locale.feature]: `Given I want to create a new locale`

File: features/localization/managing_locales/adding_locale.feature [hunk 4]
```diff
@@ -0,0 +1,16 @@
...
+        When I choose Norwegian
+        And I add it
+        Then I should be notified about success
+        And store should be available in Norwegian language
```
Comment (TheMadeleine) [inline features/localization/managing_locales/adding_locale.feature]: **the** store

File: features/localization/managing_locales/adding_locale.feature [hunk 5]
```diff
@@ -0,0 +1,16 @@
+@localization
+Feature: Adding locale
+    In order to provide service to my customers in their preferred language
+    As an Administrator
+    I want to add locale to the registry
```
Comment (TheMadeleine) [inline features/localization/managing_locales/adding_locale.feature]: *locales*? or *a locale*

File: features/localization/managing_locales/locale_unique_code_validation.feature [hunk 6]
```diff
@@ -0,0 +1,16 @@
+@localization
+Feature: Locale unique code validation
+    In order to avoid making mistakes when managing locales
+    As an Administrator
+    I want to be prevented from adding locale with existing code
```
Comment (TheMadeleine) [inline features/localization/managing_locales/locale_unique_code_validation.feature]: a locale with an existing code :D

File: features/localization/managing_locales/locale_unique_code_validation.feature [hunk 7]
```diff
@@ -0,0 +1,16 @@
...
+
+    Background:
+        Given the store is available in Norwegian language
```
Comment (TheMadeleine) [inline features/localization/managing_locales/locale_unique_code_validation.feature]: **the** Norwegian language

File: features/localization/managing_locales/locale_unique_code_validation.feature [hunk 8]
```diff
@@ -0,0 +1,16 @@
...
+        And I am logged in as administrator
+
+    @ui
+    Scenario: Trying to add new locale with used code
+        Given I want to create new locale
```
Comment (TheMadeleine) [inline features/localization/managing_locales/locale_unique_code_validation.feature]: **a** new locale

File: features/localization/managing_locales/locale_validation.feature [hunk 9]
```diff
@@ -0,0 +1,16 @@
+@localization
+Feature: Locale validation
+    In order to avoid making mistakes when managing locales
+    As an Administrator
+    I want to be prevented from adding locale without code
```
Comment (TheMadeleine) [inline features/localization/managing_locales/locale_validation.feature]: **a** locale without **a** code

File: features/localization/managing_locales/locale_validation.feature [hunk 10]
```diff
@@ -0,0 +1,16 @@
...
+
+    Background:
+        Given the store is available in Norwegian language
+        And I am logged in as administrator
+
+    @todo
+    Scenario: Trying to add locale without specifying code
+        Given I want to create new locale
```
Comment (TheMadeleine) [inline features/localization/managing_locales/locale_validation.feature]: **a** new locale

File: features/localization/managing_locales/locale_validation.feature [hunk 11]
```diff
@@ -0,0 +1,16 @@
+@localization
+Feature: Locale validation
+    In order to avoid making mistakes when managing locales
+    As an Administrator
+    I want to be prevented from adding locale without code
+
+    Background:
+        Given the store is available in Norwegian language
```
Comment (TheMadeleine) [inline features/localization/managing_locales/locale_validation.feature]: in **the** Norwegian language

File: features/localization/managing_locales/locale_validation.feature [hunk 12]
```diff
@@ -0,0 +1,16 @@
...
+        And I am logged in as administrator
+
+    @todo
+    Scenario: Trying to add locale without specifying code
```
Comment (TheMadeleine) [inline features/localization/managing_locales/locale_validation.feature]: **a** locale

File: features/localization/managing_locales/locale_validation.feature [hunk 13]
```diff
@@ -0,0 +1,16 @@
...
+        Given I want to create new locale
+        When I do not choose a code
+        And I try to add it
+        Then I should be notified that code is required
```
Comment (TheMadeleine) [inline features/localization/managing_locales/locale_validation.feature]: **a** code is required

File: features/localization/managing_locales/locale_unique_code_validation.feature [hunk 14]
```diff
@@ -0,0 +1,16 @@
...
+    In order to avoid making mistakes when managing locales
+    As an Administrator
+    I want to be prevented from adding locale with existing code
+
+    Background:
+        Given the store is available in Norwegian language
+        And I am logged in as administrator
+
+    @ui
+    Scenario: Trying to add new locale with used code
```
Comment (TheMadeleine) [inline features/localization/managing_locales/locale_unique_code_validation.feature]: **a** new locale with **an** used code.

File: features/localization/managing_locales/adding_locale.feature [hunk 15]
```diff
@@ -0,0 +1,16 @@
...
+
+    Background:
+        Given I am logged in as administrator
+
+    @ui
+    Scenario: Adding new locale
```
Comment (TheMadeleine) [inline features/localization/managing_locales/adding_locale.feature]: **a** new locale

File: features/localization/managing_locales/adding_locale.feature [hunk 16]
```diff
@@ -0,0 +1,16 @@
+@localization
+Feature: Adding locale
```
Comment (TheMadeleine) [inline features/localization/managing_locales/adding_locale.feature]: **a** locale

File: features/localization/managing_locales/locale_unique_code_validation.feature (hunk 14 above)
Comment (NeverResponse) [inline features/localization/managing_locales/locale_unique_code_validation.feature]: `a used`! `an used` is incorrect

Comment (TheMadeleine) [inline features/localization/managing_locales/locale_unique_code_validation.feature]: true. My bad :D

**Comment by pjedrzejewski** [comment]
Thanks Łukasz! We should definitely look at extracting these common steps, as soon as we have 2 cruds in a single suite = we got a problem...

**Pull request merged**